format/rtmp/rtmp.go

```diff
@@ -289,6 +289,27 @@ func (self *Conn) pollAVTag() (tag flvio.Tag, err error) {
 		case msgtypeidVideoMsg, msgtypeidAudioMsg:
 			tag = self.avtag
 			return
+		case msgtypeidDataMsgAMF0:
+			if probing {
+				onMetaDataFound := false
+				for _, val := range self.datamsgvals {
```

Comment:
Can we do this check within the `handleMsg` function so we aren't introducing a bool for `pollAVTag` ? The bool is only used during specific code paths. It's not ideal to have a control flow dependency based on what the function's caller is doing.

Reply:
Don't want to put it into `handleMsg` - it works on every message, and checking for data inside messages require working with runtime type information - which I presume is not fast.
I've made commit with slight change - moved `onMetadata` check up in the chain.

Reply:
Messages aren't sent very often (usually just a few at the beginning / end of a stream), and the runtime type lookup only needs to occur for the `onMetaData` message itself - which we have to do at some point anyway. The overall flow seems easier to reason about this way.

(We could also inline the onMetaData handling to avoid a second pass and the runtime lookup, but that's a more involved change.)

Reply:
> and the runtime type lookup only needs to occur for the onMetaData message itself

To check if it is `onMetaData` message I need, for every message iterate over `[]interface{}`, check if its value is a string, and then check if it's equal `onMetaData`.

Also, handling onMetatData in `handleMsg` would require either set properties of `self.prober` directly from `handleMsg` - which looks really wrong, or to add new data structure and new field to `Conn` object to pass parsed `onMedata` object from `handleMsg` up to `probe` func, where it is really needed.